modelkit/core/model.py

```diff
@@ -35,6 +36,34 @@
 
 logger = get_logger(__name__)
 
+ModelDependency = TypeVar(
+    "ModelDependency", bound=Union["Model", "AsyncModel", "WrappedAsyncModel"]
+)
+
+
+class ModelDependenciesMapping:
+    def __init__(
+        self, models: Dict[str, Union["Model", "AsyncModel", "WrappedAsyncModel"]]
+    ) -> None:
+        self.models = models
```

Comment:
```suggestion
    def __init__(self, models: Optional[Dict[str, ModelDependency]] = None):
        self.models = models or {}
```
If we want to allow step-by-step building as discussed below v